dot/core/service_test.go

```diff
@@ -28,6 +36,72 @@ import (
 
 var errTestDummyError = errors.New("test dummy error")
 
+func generateExtrinsic(t *testing.T) (ext types.Extrinsic, externExt types.Extrinsic, body *types.Body) {
+	rawMeta := common.MustHexToBytes(testdata.NewTestMetadata())
+	var decoded []byte
+	err := scale.Unmarshal(rawMeta, &decoded)
+	require.NoError(t, err)
+
+	meta := &ctypes.Metadata{}
+	err = ctypes.DecodeFromBytes(decoded, meta)
+	require.NoError(t, err)
```

Comment:
```suggestion
	rawMeta := common.MustHexToBytes(testdata.NewTestMetadata())
	var decoded []byte
	err := scale.Unmarshal(rawMeta, &decoded)
	require.NoError(t, err)

	meta := &ctypes.Metadata{}
	err = ctypes.DecodeFromBytes(decoded, meta)
	require.NoError(t, err)
```
this could be put in a helper.